simulation/traffic_simulator/src/vehicle_simulation/ego_entity_simulation.cpp

```diff
@@ -0,0 +1,307 @@
+  vehicle_model_type_(getVehicleModelType()),
+  vehicle_model_ptr_(makeSimulationModel(vehicle_model_type_, step_time, parameters))
+{
+}
+
+auto toString(const VehicleModelType datum) -> std::string
+{
+#define BOILERPLATE(IDENTIFIER)      \
+  case VehicleModelType::IDENTIFIER: \
+    return #IDENTIFIER
+
+  switch (datum) {
+    BOILERPLATE(DELAY_STEER_ACC);
+    BOILERPLATE(DELAY_STEER_ACC_GEARED);
+    BOILERPLATE(DELAY_STEER_VEL);
+    BOILERPLATE(IDEAL_STEER_ACC);
+    BOILERPLATE(IDEAL_STEER_ACC_GEARED);
+    BOILERPLATE(IDEAL_STEER_VEL);
+  }
+
+#undef BOILERPLATE
+
+  THROW_SIMULATION_ERROR("Unsupported vehicle model type, failed to convert to string");
+}
+
+auto EgoEntitySimulation::getVehicleModelType() -> VehicleModelType
+{
+  const auto vehicle_model_type =
+    getParameter<std::string>("vehicle_model_type", "IDEAL_STEER_VEL");
+
+  static const std::unordered_map<std::string, VehicleModelType> table{
+    {"DELAY_STEER_ACC", VehicleModelType::DELAY_STEER_ACC},
+    {"DELAY_STEER_ACC_GEARED", VehicleModelType::DELAY_STEER_ACC_GEARED},
+    {"DELAY_STEER_VEL", VehicleModelType::DELAY_STEER_VEL},
+    {"IDEAL_STEER_ACC", VehicleModelType::IDEAL_STEER_ACC},
+    {"IDEAL_STEER_ACC_GEARED", VehicleModelType::IDEAL_STEER_ACC_GEARED},
+    {"IDEAL_STEER_VEL", VehicleModelType::IDEAL_STEER_VEL},
+  };
+
+  const auto iter = table.find(vehicle_model_type);
+
+  if (iter != std::end(table)) {
+    return iter->second;
+  } else {
+    THROW_SEMANTIC_ERROR("Unsupported vehicle_model_type ", vehicle_model_type, " specified");
+  }
+}
+
+auto EgoEntitySimulation::makeSimulationModel(
+  const VehicleModelType vehicle_model_type, const double step_time,
+  const traffic_simulator_msgs::msg::VehicleParameters & parameters)
+  -> const std::shared_ptr<SimModelInterface>
+{
+  // clang-format off
+  const auto acc_time_constant   = getParameter<double>("acc_time_constant",     0.1);
+  const auto acc_time_delay      = getParameter<double>("acc_time_delay",        0.1);
+  const auto steer_lim           = getParameter<double>("steer_lim",            parameters.axles.front_axle.max_steering);  // 1.0
+  const auto steer_rate_lim      = getParameter<double>("steer_rate_lim",        5.0);
+  const auto steer_time_constant = getParameter<double>("steer_time_constant",   0.27);
+  const auto steer_time_delay    = getParameter<double>("steer_time_delay",      0.24);
+  const auto vel_lim             = getParameter<double>("vel_lim",              parameters.performance.max_speed);  // 50.0
+  const auto vel_rate_lim        = getParameter<double>("vel_rate_lim",         parameters.performance.max_acceleration);  // 7.0
+  const auto vel_time_constant   = getParameter<double>("vel_time_constant",     0.1);
+  const auto vel_time_delay      = getParameter<double>("vel_time_delay",        0.1);
+  const auto wheel_base          = getParameter<double>("wheel_base",           parameters.axles.front_axle.position_x - parameters.axles.rear_axle.position_x);
```

Comment:
format was broken.

Reply:
I've fixed the format -> 21c66fcee6984144d124f1ea06689dbdad585abe